rest_framework/fields.py

```diff
@@ -1270,6 +1273,95 @@ def to_internal_value(self, data):
+    """
+    A field that gets its representation from its parent.
+
+    This method could be used to serialize a tree structure, a linked list, or
+    even a directed acyclic graph. As with all recursive things, it is
+    important to keep the base case in mind. In the case of the tree serializer
+    example below, the base case is a node with an empty list of children. In
+    the case of the list serializer below, the base case is when `next==None`.
+    Above all, beware of cyclical references.
+
+    Examples:
+
+    class TreeSerializer(self):
+        children = ListField(child=RecursiveField())
+
+    class ListSerializer(self):
+        next = RecursiveField(allow_null=True)
+    """
+
+    # This list of attributes determined by the attributes that
+    # `rest_framework.serializers` calls to on a field object
+    PROXIED_ATTRS = (
+        # methods
+        'get_value',
+        'get_initial',
+        'run_validation',
+        'get_attribute',
+        'to_representation',
+
+        # attributes
+        'field_name',
+        'source',
+        'read_only',
+        'default',
+        'source_attrs',
+        'write_only',
+    )
+
+    def __init__(self, to=None, **kwargs):
+        """
+        arguments:
+        to - `None`, the name of another serializer defined in the same module
+             as this serializer, or the fully qualified import path to another
+             serializer. e.g. `ExampleSerializer` or
+             `path.to.module.ExampleSerializer`
+        """
+        self.to = to
+        self.kwargs = kwargs
+
+    def bind(self, field_name, parent):
+        if hasattr(parent, 'child') and parent.child is self:
+            # RecursiveField nested inside of a ListField
+            parent_class = parent.parent.__class__
+        else:
+            # RecursiveField directly inside a Serializer
+            parent_class = parent.__class__
+
+        if self.to is None:
+            proxied_class = parent_class
+        else:
+            try:
+                module_name, class_name = self.to.rsplit('.', 1)
+            except ValueError:
+                module_name, class_name = parent_class.__module__, self.to
+
+            try:
+                proxied_class = getattr(
+                    importlib.import_module(module_name), class_name)
+            except Exception as e:
+                raise ImportError(
+                    'could not locate serializer %s' % self.to, e)
+
+        # Create a new serializer instance and proxy it
+        proxied = proxied_class(**self.kwargs)
+        proxied.bind(field_name, parent)
+        self.proxied = proxied
+
+    def __getattribute__(self, name):
+        if name in RecursiveField.PROXIED_ATTRS:
+            try:
+                proxied = object.__getattribute__(self, 'proxied')
+                return getattr(proxied, name)
+            except AttributeError:
+                pass
+
+        return object.__getattribute__(self, name)
```

Comment:
```
class Service(MPTTModel):
    name = models.CharField(max_length=250)
    parent = TreeForeignKey('self', blank=True, null=True, related_name='children')

class ServiceSerializer(serializers.ModelSerializer):
    parent = serializers.RecursiveField(required=False)
    class Meta:
        model = Service
        fieds = ('id', 'name', 'parent')

# Gets called here
class LawyerSerializer(serializers.ModelSerializer):
    markets = MarketSerializer(many=True)
    services = ServiceSerializer(many=True)
```

I've tried specifying source as 'parent', and also tried it on a regular foreign key to 'self' rather than the MPTT TreeForeignKey on the model.  I've also verified that I have the entire patch applied (I checked out your brancch for the time being until its merged).  thanks for any help @heywbj 

and the exact error:
ipdb> object.__getattribute__(self, name)
*** AttributeError: 'RecursiveField' object has no attribute 'source'